src/filter/exceptions.filter.ts

```diff
@@ -0,0 +1,27 @@
+import { ExceptionFilter, Catch, ArgumentsHost, HttpException, UnauthorizedException, BadRequestException } from '@nestjs/common';
+import { Response } from 'express';
+import { ErrorResponse } from '../dto/error-response.dto';
+
+@Catch()
+export class ExceptionsFilter implements ExceptionFilter {
+  catch(exception: any, host: ArgumentsHost) {
+    const ctx = host.switchToHttp();
+    const response = ctx.getResponse<Response>();
+    let status = exception instanceof HttpException ? exception.getStatus() : 500;
```

Comment:
It is not clear what 500 is. We can change it as follows. 

`import { HttpStatus, HttpException } from '@nestjs/common'; 

Let status = exception instanceof HttpException
        ? exception.getStatus()
        : HttpStatus.INTERNAL_SERVER_ERROR;
`